### PR TITLE
Allow smell detectors to be specified by category

### PR DESCRIPTION
**Pull request by mvz**
#863

### CODE REVIEW

File: features/command_line_interface/options.feature [hunk 1]
```diff
@@ -41,7 +41,8 @@ Feature: Reek can be controlled using command-line options
 
       Configuration:
           -c, --config FILE                Read configuration options from FILE
-              --smell SMELL                Detect smell SMELL (default: all enabled smells)
+              --smell SMELL                Detect smell or smell category SMELL
```
Comment (troessner) [inline features/command_line_interface/options.feature]: >> Detect smell or smell category SMELL

Can we remove the SMELL here? Seems unnecessary and reads pretty bad IMHO.

File: lib/reek/cli/options.rb [hunk 2]
```diff
@@ -92,7 +92,9 @@ def set_configuration_options
           raise ArgumentError, "Config file #{file} doesn't exist" unless File.exist?(file)
           self.config_file = Pathname.new(file)
         end
-        parser.on('--smell SMELL', 'Detect smell SMELL (default: all enabled smells)') do |smell|
+        parser.on('--smell SMELL',
+                  'Detect smell or smell category SMELL',
```
Comment (troessner) [inline lib/reek/cli/options.rb]: Same here.....

File: spec/reek/smells/smell_repository_spec.rb [hunk 3]
```diff
@@ -19,4 +19,21 @@
...
+    it 'returns smell detectors specified by type' do
+      result = described_class.eligible_smell_types(['IrresponsibleModule', 'TooManyStatements'])
+      expect(result).to eq [Reek::Smells::IrresponsibleModule,
+                            Reek::Smells::TooManyStatements]
+    end
+
+    it 'returns smell detectors specified by category' do
+      result = described_class.eligible_smell_types(['Duplication'])
+      expect(result).to eq [Reek::Smells::DuplicateMethodCall]
+    end
```
Comment (troessner) [inline spec/reek/smells/smell_repository_spec.rb]: Nice specs, clean and concise.

File: features/command_line_interface/options.feature [hunk 4]
```diff
@@ -41,7 +41,8 @@ Feature: Reek can be controlled using command-line options
...
+                                             (default: all enabled smells)
```
Comment (troessner) [inline features/command_line_interface/options.feature]: Also, can we add some examples for both (smell types and smell categories) to our README including helpful links to our docs so the user can explore that on its own?
If we just put it out like this I believe the mental barrier for the user is so high that nobody except for maybe us will use it...

**Comment by troessner** [comment]
2 small comment and then lgtm!

File: features/command_line_interface/options.feature (hunk 1 above)
Comment (mvz) [inline features/command_line_interface/options.feature]: Well, we need to put something to indicate that a parameter is required. Suggestions?

Comment (troessner) [inline features/command_line_interface/options.feature]: Hmm, how about adding <>, so <SMELL> to make it more obvious?
Right now it reads like a typo.

**Comment by mvz** [comment]
I was updating the documentation and noticed that LongParameterList is both a smell type and a smell category that also includes LongYieldList. We should fix that before exposing categories like this (since there would be no way to only detect the *smell* LongParameterList).

**Comment by troessner** [comment]
This one is now obsolete with #867 isn't it?

**Comment by mvz** [comment]
Yes, this one is obsolete.

**Pull request closed without merging**